app.rb

```diff
@@ -0,0 +1,42 @@
+require 'sinatra/base'
+require './lib/peep'
+require './lib/account'
+
+class ChitterChallenge < Sinatra::Base
+  enable :sessions 
+
+  get '/' do
+    'Welcome to Chitter!'
+  end
+
+  get '/signup' do
+    erb :'signup'
+  end
+
+  post '/peeps/signed_new' do
+    account = Account.create(forename: params[:forename], surname: params[:surname], username: params[:username], email: params[:email], password: params[:password])
+    session[:username] = account.username
+    redirect '/peeps/new'
+  end
+
+  get '/peeps' do
+    @peeps = Peep.all
+
+    erb :'peeps/index'
+  end
+
+  get '/peeps/new' do
+
+    erb :'peeps/new'
+  end
+
+  post '/peeps' do
+    session[:username] != nil ? username = session[:username] : username = 'Guest'
```

Comment:
what is this doing? Is it letting to post a peep even if you don't have an account?